packages/controller/main.js

```diff
@@ -3014,73 +3021,68 @@ function checkVersion(id, name, version, instances) {
+ * @param {string[]|object[]|string} globalDeps - global dependencies, as defined in io-pack
+ * @returns {Promise<void>}
+ */
+async function checkVersions(id, deps, globalDeps) {
+    const res = await objects.getObjectViewAsync('system', 'instance', {startkey: 'system.adapter.', endkey: 'system.adapter.\u9999'});
+    const instances = {};
+    const globInstances = {};
+    if (res && res.rows) {
+        res.rows.forEach(item => {
+            if (!item.value._id) {
+                return;
             }
-
-            // this ensures we have a real object with correct structure
-            deps = tools.parseDependencies(deps);
-            globalDeps = tools.parseDependencies(globalDeps);
-
-            // check local dependencies: required adapter must be installed on the same host
-            try {
-                for (const dep of Object.keys(deps)) {
-                    if (!checkVersion(id, dep, deps[dep], instances)) {
-                        return reject(new Error());
-                    }
-                }
-            } catch (e) {
-                logger.error(`${hostLogPrefix} startInstance ${id} [checkVersions]: ${e}`);
-                logger.error(`${hostLogPrefix} startInstance ${id} [checkVersions]: ${JSON.stringify(deps)}`);
-                return reject(new Error());
+            globInstances[item.value._id] = item.value;
+        });
+        Object.keys(globInstances).forEach(id => {
+            if (globInstances[id] && globInstances[id].common && globInstances[id].common.host === hostname) {
+                instances[id] = globInstances[id];
             }
+        });
+    }
 
-            // check global dependencies: required adapter must be NOT installed on the same host
-            try {
-                for (const gDep of Object.keys(globalDeps)) {
-                    if (!checkVersion(id, gDep, globalDeps[gDep], globInstances)) {
-                        return reject(new Error());
-                    }
-                }
-            } catch (e) {
-                logger.error(`${hostLogPrefix} startInstance ${id} [checkVersions]: ${e}`);
-                logger.error(`${hostLogPrefix} startInstance ${id} [checkVersions]: ${JSON.stringify(globalDeps)}`);
-                return reject(new Error());
-            }
+    // this ensures we have a real object with correct structure
+    deps = tools.parseDependencies(deps);
+    globalDeps = tools.parseDependencies(globalDeps);
 
-            resolve();
-        });
-    });
+    // check local dependencies: required adapter must be installed on the same host
+    try {
+        for (const dep of Object.keys(deps)) {
+            checkVersion(dep, deps[dep], instances);
+        }
+    } catch (e) {
+        logger.error(`${hostLogPrefix} startInstance ${id} [sameHostDependency]: ${e.message}`);
+        logger.debug(`${hostLogPrefix} startInstance ${id} [sameHostDependency]: ${JSON.stringify(deps)}`);
+        throw new Error();
```

Comment:
Empty error? why not simply rethrow original one?

Reply:
Was like this before but we can change